# Type: test
# Subtype: T1
# English Summary: Add test cases for NamespaceGenTest
# Chinese Summary: 为NamespaceGenTest添加测试用例
# Entry Number: 28

# === Git Diff Content ===

diff --git a/idea/tests/org/jetbrains/jet/codegen/NamespaceGenTest.java b/idea/tests/org/jetbrains/jet/codegen/NamespaceGenTest.java
index 76dedc0..a0c46ba 100644
--- a/idea/tests/org/jetbrains/jet/codegen/NamespaceGenTest.java
+++ b/idea/tests/org/jetbrains/jet/codegen/NamespaceGenTest.java
@@ -530,6 +530,13 @@ public class NamespaceGenTest extends LightCodeInsightFixtureTestCase {
         assertEquals(Boolean.FALSE, main.invoke(null, null, "jet"));
     }
 
+    public void testEqualsNullLiteral() throws Exception {
+        loadText("fun foo(s: String?) = s == null");
+        final Method main = generateFunction();
+        assertEquals(Boolean.TRUE, main.invoke(null, new Object[] { null }));
+        assertEquals(Boolean.FALSE, main.invoke(null, "jet"));
+    }
+
     public void testFunctionCall() throws Exception {
         loadFile("functionCall.jet");
         System.out.println(generateToText());
